src/main/java/hudson/plugins/ec2/EC2SpotSlave.java

```diff
@@ -178,7 +180,7 @@ protected boolean isAlive(boolean force) {
      * Cancel the spot request for the instance. Terminate the instance if it is up. Remove the agent from Jenkins.
      */
     @Override
-    public void terminate() {
+    protected void _terminate(TaskListener listener) throws IOException, InterruptedException {
```

Comment:
The impl seems mostly the same as in `EC2OndemandSlave`. Is there a reason the method cannot simply be pulled up into `EC2AbstractSlave`.